display_list/display_list_builder.cc

```diff
@@ -237,7 +237,13 @@ void DisplayListBuilder::restore() {
     layer_stack_.pop_back();
     current_layer_ = &layer_stack_.back();
     Push<RestoreOp>(0, 1);
-    if (!layer_info.has_layer) {
+    if (layer_info.has_layer) {
+      if (layer_info.is_group_opacity_compatible()) {
+        SaveLayerOp* op = reinterpret_cast<SaveLayerOp*>(
+            storage_.get() + layer_info.save_layer_offset);
+        op->options = op->options.with_can_distribute_opacity();
```

Comment:
This is a little terrifying to me. Maybe we should add a warning here that we should make sure we only ever modify data in-place? ie - that nothing we do here increases the size of *op.

Reply:
And that this type of modification can only be done during building. Once done with building the contents of the records must remain constant so we can trust its output to never change and Equals() to continue working.

Reply:
Big block comment added...